odoo/modules/module.py

```diff
@@ -311,74 +351,85 @@ def get_module_root(path):
-        # automatically installed if all dependencies are (special case: [] to
-        # always install the module), either `True` to auto-install the module
-        # in case all dependencies declared in `depends` are installed.
-        if isinstance(info['auto_install'], collections.abc.Iterable):
-            info['auto_install'] = set(info['auto_install'])
-            non_dependencies = info['auto_install'].difference(info['depends'])
-            assert not non_dependencies,\
-                "auto_install triggers must be dependencies, found " \
-                "non-dependencies [%s] for module %s" % (
-                    ', '.join(non_dependencies), module
-                )
-        elif info['auto_install']:
-            info['auto_install'] = set(info['depends'])
-
-        info['version'] = adapt_version(info['version'])
-        return info
-
-    _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
-    return {}
+
+    if not manifest_file:
+        _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
+        return {}
+
+    manifest = copy.deepcopy(_DEFAULT_MANIFEST)
+    manifest['icon'] = get_module_icon(module)
+
+    f = tools.file_open(manifest_file, mode='rb')
+    try:
+        manifest.update(ast.literal_eval(pycompat.to_text(f.read())))
+    finally:
+        f.close()
+
+    if not manifest.get('description'):
+        readme_path = [opj(mod_path, x) for x in README
+                       if os.path.isfile(opj(mod_path, x))]
+        if readme_path:
+            with tools.file_open(readme_path[0]) as fd:
+                manifest['description'] = fd.read()
+
+    if not manifest.get('license'):
+        manifest['license'] = 'LGPL-3'
+        _logger.warning("Missing `license` key in manifest for %r, defaulting to LGPL-3", module)
+    if manifest.get('auto-install'):
+        _logger.warning("Wrong 'auto-install' key in manifest for %r, assuming 'auto_install'", module)
+        manifest['auto_install'] = manifest.pop('auto-install')
+    if manifest.get('autoinstall'):
+        _logger.warning("Wrong 'autoinstall' key in manifest for %r, assuming 'auto_install'", module)
+        manifest['auto_install'] = manifest.pop('autoinstall')
+    if 'qweb' in manifest:
+        _logger.warning("Ignored 'qweb' key in manifest for %r, you should read about asset bundles. See https://www.odoo.com/documentation/15.0/developer/reference/frontend/assets.html", module)
+    if 'css' in manifest:
+        _logger.warning("Ignored 'css' key in manifest for %r, you should read about asset bundles. See https://www.odoo.com/documentation/15.0/developer/reference/frontend/assets.html", module)
+    if 'js' in manifest:
+        _logger.warning("Ignored 'js' key in manifest for %r, you should read about asset bundles. See https://www.odoo.com/documentation/15.0/developer/reference/frontend/assets.html", module)
+
```

Comment:
Let's please stop bloating the codebase with useless warnings, especially those that can be triggered 200x per registry loading.

The framework is not a spellchecker either. what about `auto-instal` and `auto_intsall`?

Besides, there are legit cases where maintainers need to keep compatibility with multiple versions of the framework at once, and we're plaguing them with useless warnings.

I am 100% in favor of warnings that are helping developers catch tricky bugs that will take them hours to debug. But this is not the case here.

Reply:
I totally agree with you, dropping those useless warnings.

Reply:
@odony this came from the discussion on https://github.com/odoo/odoo/pull/80988